Kinvey.Core/Offline/SQLiteCacheManager.cs

```diff
@@ -207,7 +213,48 @@ public List<string> getCollectionTables()
 			return collections;
 		}
 
-		public ISyncQueue GetSyncQueue(string collectionName) {
+        public QueryCacheItem GetQueryCacheItem(string collectionName, string query, string lastRequest)
+        {
+            QueryCacheItem result = null;
+
+            if (!TableExists<QueryCacheItem>(DBConnectionSync))
+            {
+                DBConnectionSync.CreateTable<QueryCacheItem>();
+            }
+            else
+            {
+                var items = DBConnectionSync.Table<QueryCacheItem>().Where(item => item.collectionName == collectionName && item.query == query);
+                if (items.Count() == 1)
```

Comment:
that's why I was looking to see if we can make the combination of `collectionName` and `query` unique so we could do something like `Where(...).First` and make sure there's only one or none record

Reply:
Yes, would be convenient, but unfortunately not an option.